packages/core/src/lib/GooglePlay.ts

```diff
@@ -62,12 +65,110 @@ export class GooglePlay {
    * https://github.com/Triple-T/gradle-play-publisher#uploading-a-pre-existing-artifact
    * @param track - Specifies the track that the user would like to publish to.
    */
-  async publishBundle(track: PlayStoreTrack, filepath: string): Promise<void> {
+  async publishBundleWithGradle(track: PlayStoreTrack, filepath: string): Promise<void> {
     // Uploads the artifact to the default internal track.
     await this.gradleWrapper.executeGradleCommand(
         ['publishBundle', '--artifact-dir', filepath, '--track', track]);
   }
 
+  /**
+   * This calls the publish bundle command and publishes an existing artifact to Google
+   * Play.
+   * Calls the following Play API commands in order:
+   * Edits.Insert
+   * Edits.Bundles.Upload
+   * Edits.Tracks.Update
+   * Edits.Commit
+   * https://developers.google.com/android-publisher/api-ref/rest/v3/edits.bundles/upload
+   * https://developers.google.com/android-publisher/edits#workflow
+   * @param track - Specifies the track that the user would like to publish to.
+   * @param filepath - Filepath of the App bundle you would like to upload.
+   * @param packageName - Package name of the bundle.
+   * @param retainedBundles - All bundles that should be retained on upload. This is useful for
+   *   ChromeOS only releases.
+   */
+  async publishBundle(
+      track: PlayStoreTrack,
+      filepath: string,
+      packageName: string,
+      retainedBundles: number[],
+  ): Promise<void> {
+    // TODO(@nohe427): Remove this check when refactor is finished.
+    if (!this._googlePlayApi) {
+      throw new Error('Service Account JSON file not set.');
+    }
+    const edit = await this._googlePlayApi.edits.insert({packageName: packageName});
+    const editId = edit.data.id;
+    if (!editId) {
+      throw new Error('Could not create a Google Play edit');
```

Comment:
For a follow up CL - do we know why this could fail? Should we retry or should we just tell the user "sorry, that didn't work, try again later"?

Reply:
I am not sure why it would potentially fail just yet. I am going to add a TODO here.